test/unit/database_tables.js

```diff
@@ -194,4 +194,288 @@ describe('DatabaseTables', function() {
+            },
+            {
+                result: [
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tableone', updated_at: undefined},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tabletwo'},
+                    {
+                        dbname: 'db1',
+                        schema_name: 'public',
+                        table_name: 'analysis_b194a8f896_81cc00c1cfbd5c04d3375fc0e0343a34ae979f34'
+                    }
+                ],
+                skipNotUpdatedAtTables: true,
+                skipAnalysisCachedTables: true,
+                expectedLength: 0
+            },
+            {
+                result: [
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tableone', updated_at: Date.now()},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tabletwo'},
+                    {
+                        dbname: 'db1',
+                        schema_name: 'public',
+                        table_name: 'analysis_b194a8f896_81cc00c1cfbd5c04d3375fc0e0343a34ae979f34',
+                        updated_at: Date.now()
+                    }
+                ],
+                skipNotUpdatedAtTables: true,
+                skipAnalysisCachedTables: true,
+                expectedLength: 1
+            },
+            {
+                result: [
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tableone', updated_at: Date.now()},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tabletwo', updated_at: Date.now()},
+                    {
+                        dbname: 'db1',
+                        schema_name: 'public',
+                        table_name: 'analysis_b194a8f896_81cc00c1cfbd5c04d3375fc0e0343a34ae979f34',
+                        updated_at: Date.now()
+                    }
+                ],
+                skipNotUpdatedAtTables: true,
+                skipAnalysisCachedTables: true,
+                expectedLength: 2
+            },
+            {
+                result: [
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tableone', updated_at: Date.now()},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tabletwo'},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tablethree', updated_at: Date.now()},
+                    {
+                        dbname: 'db1',
+                        schema_name: 'public',
+                        table_name: 'analysis_b194a8f896_81cc00c1cfbd5c04d3375fc0e0343a34ae979f34',
+                        updated_at: Date.now()
+                    }
+                ],
+                skipNotUpdatedAtTables: true,
+                skipAnalysisCachedTables: true,
+                expectedLength: 2
+            }
+        ];
+
+        scenarios.forEach(function ({ result, skipNotUpdatedAtTables, skipAnalysisCachedTables, expectedLength }) {
```

Comment:
I don't particularly enjoy this type of testing, it took me a while to understand what was going on, but at least is complete which is pretty nice.

Reply:
I've just followed the way tests are implemented in this project, this approach is widely used in our Node.js stack by the way. I'm not a big fan I must admit, tests should be as simpler as possible.